unmock/core/http.py

```diff
@@ -0,0 +1,143 @@
+from typing import Optional, List
+import http.client
+from urllib.parse import urlsplit, SplitResult
+
+from .options import UnmockOptions
+from .utils import Patchers
+
+__all__ = ["initialize", "reset"]
+
+# Backup:
+UNMOCK_AUTH = "___u__n_m_o_c_k_a_u_t__h_"
+PATCHERS = Patchers()
+STORIES = list()
+
+def parse_url(url) -> SplitResult:
+    parsed_url = urlsplit(url)
+    if parsed_url.scheme == "" or parsed_url.netloc == "":
+        return urlsplit("http://{url}".format(url=url))
```

Comment:
Do we need to add the protocol if none is given? Nowadays https seems more like the default to me :)

Reply:
Unfortunately, yes we need to add the protocol for `urlsplit` to work properly. I'll add a comment, but it does not matter if it's `http` or `https`, it's just used internally for us :)